docs/walkthrough/using-callbacks.md

```diff
@@ -0,0 +1,53 @@
+[14:10:40] INFO     Done ✨                                                                              __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 3 minutes and 38 seconds                             __main__.py:196
+           DEBUG    Metric: 'resnet50_average_precision' Value: 0.16654                                  __main__.py:205
+           DEBUG    Metric: 'resnet50_dcg_at_k' Value: 0.24018                                           __main__.py:205
+           DEBUG    Metric: 'resnet50_f1_score_at_k' Value: 0.03631                                      __main__.py:205
+           DEBUG    Metric: 'resnet50_hit_at_k' Value: 0.38123                                           __main__.py:205
+           DEBUG    Metric: 'resnet50_ndcg_at_k' Value: 0.24018                                          __main__.py:205
+           DEBUG    Metric: 'resnet50_precision_at_k' Value: 0.01906                                     __main__.py:205
+           DEBUG    Metric: 'resnet50_r_precision' Value: 0.16654                                        __main__.py:205
+           DEBUG    Metric: 'resnet50_recall_at_k' Value: 0.38123                                        __main__.py:205
+           DEBUG    Metric: 'resnet50_reciprocal_rank' Value: 0.16654                                    __main__.py:205
+           INFO     Building the artifact ...                                                            __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                                       __main__.py:231
+```
+
+The evaluation callback is triggered at two stages: at the start of fitting to generate progress bars to track the evaluation progress, and at the end of each epoch, in which the model is evaluated using the query and index datasets that were provided when callback was created.
+
+## Early Stopping
+
+The early stopping callback measures a given metric at the end of every epoch, and will stop the fitting early if the metric doesnt increase at a high enough rate. The best performing model out of the completed epochs is stored in the case that it starts to perform worse after a certain point. Both the minimum improvement and the number of successive rounds without improvement before the training ends can be specified as optional arguments at construction.
+The example output below shows a situation in which the early stopping callback is used to end a training run early. Compared to the results above in the evaluation explanation, the number of epochs was only two, compared to the full five, more than halfing the runtime of the training. On the other hand, the final loss recorded is 0.007, more than twice as high as if all five training epochs had completed. While the parameters of the early stopping callback can be adjusted to further improve either the runtime or loss, however this tradeoff between the two outcomes is what the early stopping callback promises to deliver.
+
```

Comment:
add some code block here demonstrate the parameters and default ones and how to use them